<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>
        let p1 = new Promise((resolve, reject) => {
            console.log("Promise1 is pending");
            setTimeout(() => {
                console.log('I am a promise1 i am fullfilled')
                resolve(true)
            }, 5000)
        })

        let p2 = new Promise((resolve, reject) => {
            console.log("Promise2 is pending");
            setTimeout(() => {
                console.log('I am a promise2 i am rejected')
                reject(new Error("I am an error"))
            }, 5000)
        })

        console.log(p1, p2);

        // after p1 execution do 
        p1.then((value) => {
            console.log(value);
        })

        // after p2 execution do not execute beacuse p2 is rejected
        //p2.then((value)=>{
        //  console.log(value);
        //})

        // after error or reject catch it
        p2.catch((error) => {
            console.log("some error occured in p2 ");
        })

        // or for catching use this
        p2.then((value) => {
            console.log(value);
        },(error)=>{
            console.log(error);
        })

    </script>
</body>

</html>